### PR TITLE
Fix for pbs_tmrsh invoked from sister moms failure

### PR DESCRIPTION
**Pull request by sujatapatnaik52**
#### Describe Bug or Feature
pbs_tmrsh invoked from sister moms fails with "cannot find PBS_NODEFILE".

#### Describe Your Change
1. PBS_NODEFILE environment variable to be set in sister mom. 
2. PBS_NODEFILE to be created in sister moms. 
3. The above change is enabled by default. For disabling it, a new boolean mom config parameter is introduced i.e. $secondary_pbs_nodefile

After making the above changes, the invocation was still failing with this error message - "start_process, no stream to MS". Explicitly initializing the mother superior's stream resolves this. 

#### Attach Test and Valgrind Logs/Output

[TestSecondaryPBSNodefile.txt](https://github.com/openpbs/openpbs/files/5640835/TestSecondaryPBSNodefile.txt)
[TestJobTask.txt](https://github.com/openpbs/openpbs/files/5640836/TestJobTask.txt)

### CODE REVIEW

File: test/tests/functional/pbs_job_task.py [hunk 1]
```diff
@@ -103,3 +103,41 @@ def test_singlenode_pbs_tmrsh(self):
...
         self.check_jobs_file(job_output_file)
+
+    def test_invoke_pbs_tmrsh_from_sister_mom(self):
+        """
+        This test cases verifies pbs_tmrsh invoked from sister mom
+        executes successfully
+        """
+        # Skip test if number of mom provided is not equal to three
+        if len(self.moms) != 3:
+            self.skipTest("test requires atleast three MoMs as input, " +
```
Comment (prakashcv13) [inline test/tests/functional/pbs_job_task.py:114]: checking for exactly 3, so we can remove "atleast"

File: test/tests/functional/pbs_secondary_pbs_nodefile.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    sister moms of job.
+    """
+    def test_secondary_pbs_nodefile_default(self):
+        """
+        This test case verifies PBS_NODEFILE gets created on
+        sister mom by default
+        """
+        # Skip test if number of mom provided is not equal to two
+        if len(self.moms) != 2:
+            self.skipTest("test requires atleast two MoMs as input, " +
```
Comment (prakashcv13) [inline test/tests/functional/pbs_secondary_pbs_nodefile.py:56]: here as well - we can remove "atleast"

**Review by prakashcv13 — CHANGES_REQUESTED** [review]

**Review by prakashcv13 — APPROVED** [review]

File: src/resmom/catch_child.c [hunk 3]
```diff
@@ -106,6 +106,7 @@ extern int		server_stream;
 extern time_t		time_now;
 extern pbs_list_head	mom_polljobs;
 extern unsigned int	pbs_mom_port;
+extern int secondary_pbs_nodefile;
```
Comment (subhasisb) [inline src/resmom/catch_child.c:109]: I think the name is confusing - secondary_pbs_nodefile can be confused with what we doing in other hook

Comment (subhasisb) [reply src/resmom/catch_child.c:109]: also is this aligned? If so, please align the other variables as well

File: src/resmom/catch_child.c [hunk 4]
```diff
@@ -1365,9 +1366,10 @@ del_job_resc(job *pjob)
 			exit(99);	/* simulate crash */
 	}
 
-	/* remove PBS_NODEFILE - Mother Superior only has one */
+	/* remove PBS_NODEFILE - Mother Superior shall have one and the sister
+	moms too if the mom config secondary_pbs_nodefile is set to 1 */
 
-	if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_HERE) {
+	if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_HERE || secondary_pbs_nodefile) {
```
Comment (subhasisb) [inline src/resmom/catch_child.c:1372]: i would think that a better name might be something like "gen_nodefile_on_sister_mom"?

File: src/resmom/mom_comm.c [hunk 5]
```diff
@@ -110,6 +110,7 @@ extern	int		server_stream;
 extern	char		mom_short_name[];
 extern unsigned int	pbs_mom_port;
 extern unsigned int	pbs_rm_port;
+extern int secondary_pbs_nodefile;
```
Comment (subhasisb) [inline src/resmom/mom_comm.c:113]: indent?

**Review by subhasisb — CHANGES_REQUESTED** [review]

**Comment by sujatapatnaik52** [comment]
[TestGenNodefileOnSisterMom.txt](https://github.com/openpbs/openpbs/files/5656769/TestGenNodefileOnSisterMom.txt)
[TestJobTask.txt](https://github.com/openpbs/openpbs/files/5656770/TestJobTask.txt)

**Comment by sujatapatnaik52** [comment]
@subhasisb Please review the latest commits.

File: src/resmom/catch_child.c [hunk 6]
```diff
@@ -106,14 +106,14 @@ extern int		server_stream;
...
-extern int secondary_pbs_nodefile;
+extern int 		gen_nodefile_on_sister_mom;
 #if MOM_ALPS
 extern useconds_t	alps_release_wait_time;
 extern int		alps_release_timeout;
 extern useconds_t	alps_release_jitter;
 #endif
 
-extern char		*path_hooks_workdir;
+extern char		*		path_hooks_workdir;
```
Comment (subhasisb) [inline src/resmom/catch_child.c]: now there is a tab between * and path_hooks_workdir

File: src/resmom/mom_main.c [hunk 7]
```diff
@@ -474,7 +474,7 @@ static handler_ret_t	set_restart_transmogrify(char *);
 static handler_ret_t	set_restrict_user(char *);
 static handler_ret_t	set_restrict_user_maxsys(char *);
 static handler_ret_t	set_restrict_user_exceptions(char *);
-static handler_ret_t	set_secondary_pbs_nodefile(char *);
+static handler_ret_t	set_gen_nodefile_on_sister_mom(char *);
```
Comment (subhasisb) [inline src/resmom/mom_main.c:477]: per our updated coding standards, variables and declarations need not be aligned - so you need to remove the alignment of all the blocks u touch

File: src/resmom/mom_comm.c [hunk 8]
```diff
@@ -110,7 +110,7 @@ extern	int		server_stream;
 extern	char		mom_short_name[];
 extern unsigned int	pbs_mom_port;
 extern unsigned int	pbs_rm_port;
-extern int secondary_pbs_nodefile;
+extern int 		gen_nodefile_on_sister_mom;
```
Comment (subhasisb) [inline src/resmom/mom_comm.c:113]: per our updated coding standards, variables and declarations need not be aligned - so you need to remove the alignment of all the blocks u touch

File: src/resmom/catch_child.c [hunk 9]
```diff
@@ -106,14 +106,14 @@ extern int		server_stream;
 extern time_t		time_now;
 extern pbs_list_head	mom_polljobs;
 extern unsigned int	pbs_mom_port;
-extern int secondary_pbs_nodefile;
+extern int 		gen_nodefile_on_sister_mom;
```
Comment (subhasisb) [inline src/resmom/catch_child.c:109]: per our updated coding standards, variables and declarations need not be aligned - so you need to remove the alignment of all the blocks u touch

**Review by subhasisb — CHANGES_REQUESTED** [review]

**Review by subhasisb — APPROVED** [review]

**Comment by sujatapatnaik52** [comment]
One of the travis builds failed, force-pushing to restart it.

**Pull request merged**